### DISCUSSION TITLE
Question about PngEncoderCore.SanitizeAndSetEncoderOptions

### DISCUSSION THREAD

**Discussion by DanTravison** [SixLabors/ImageSharp#2664] [Q&A]
I've been experimenting with various approaches to minimizing PNG file sizes and have a question about a PngFilterMethod.

It appears that PngEncoderCore.SanitizeAndSetEncoderOptions ignores encoder.FilterMethod when set.
```
if (!encoder.FilterMethod.HasValue)
{
    // Specification recommends default filter method None for paletted images and Paeth for others.
    this.filterMethod = this.colorType is PngColorType.Palette ? PngFilterMethod.None : PngFilterMethod.Paeth;
}
```
Shouldn't there be an else clause that either uses encoder.FilterMethod or uses it when colorType is no Palette?
Something like the following:
```
else
{
    this.filterMethod = this.colorType is PngColorType.Palette ? PngFilterMethod.None : encoder.FilterMethod.Value;
}
```
Am I missing something here?

**Comment by JimBobSquarePants** [comment]
You're not missing anythingactually. We should respect the user choice but fall back to a default of None for palette images. 

The png specification actually recommends no filtering for palette images as that will always yeild the best result in that case.

**Reply by DanTravison** [reply]
Perhaps this should be another discussion or is outside the area for this library so let me know.

I've been researching methods to automate a manual process.

I have a set of hand drawn images using colored pencils.  I currently load them into PaintShopPro, reduce the colors to 256 and save them as PNG.  Visually the results are quite good for color quantization and file size reduction is around 80%. What I also found is the images don't appear to use a palette (debugging OngDecoderCore never hit either of the PngChunkType.Palette case statements.)

When I use ImageSharp with the Wu quantizer and ColorType.Palette,  the best I can get is 51% reduction.

I can't seem to get anywhere near either result using no quantizer and any of the compression methods, including adaptive. 

Am I missing something or is PSP doing something custom that is also compliant.

Thanks,
Dan.
